encoding/protobq/decode_test.go

```diff
@@ -0,0 +1,194 @@
+package protobq
+
+import (
+	"testing"
+
+	"cloud.google.com/go/bigquery"
+	expr "google.golang.org/genproto/googleapis/api/expr/v1beta1"
+	"google.golang.org/genproto/googleapis/example/library/v1"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/testing/protocmp"
+	"gotest.tools/v3/assert"
+)
+
+func TestUnmarshalOptions_Unmarshal(t *testing.T) {
+	for _, tt := range []struct {
+		name     string
+		row      map[string]bigquery.Value
+		opt      UnmarshalOptions
+		expected proto.Message
+	}{
+		{
+			name: "library.Book",
+			row: map[string]bigquery.Value{
+				"name":   "name",
+				"author": "author",
+				"title":  "title",
+				"read":   true,
+			},
+			expected: &library.Book{
+				Name:   "name",
+				Author: "author",
+				Title:  "title",
+				Read:   true,
+			},
+		},
+		{
+			name: "library.UpdateBookRequest",
+			row: map[string]bigquery.Value{
+				"name": "name",
+				"book": map[string]bigquery.Value{
+					"name":   "name",
+					"author": "author",
+					"title":  "title",
+					"read":   true,
+				},
+			},
+			expected: &library.UpdateBookRequest{
+				Name: "name",
+				Book: &library.Book{
+					Name:   "name",
+					Author: "author",
+					Title:  "title",
+					Read:   true,
+				},
+			},
+		},
+		{
+			name: "expr.Value (bool)",
+			row: map[string]bigquery.Value{
+				"bool_value": true,
+			},
+			expected: &expr.Value{
+				Kind: &expr.Value_BoolValue{
+					BoolValue: true,
+				},
+			},
+		},
+		{
+			name: "expr.Value (bool)",
+			row: map[string]bigquery.Value{
+				"double_value": float64(42),
+			},
+			expected: &expr.Value{
+				Kind: &expr.Value_DoubleValue{
+					DoubleValue: 42,
+				},
+			},
+		},
```

Comment:
I'm missing tests that makes use of different unmarshal options.

Reply:
Done!